### PR TITLE
feature: add getWidgetProps, getFormatterProps to Field

### PR DESCRIPTION
**Pull request by davecoates**
This provides way for component specific props to make it's way from Field definition
to widget or formatter components.

This serves to primary purposes:
1) Allow a field like CharField to pass 'maxLength' to the component without getWidgetForField
   having to know the prop exists (previous solution).
2) Make it easy for end user to pass through arbitrary props when defining the model
   rather than having to pass it at each form/formatter usage.

### CODE REVIEW

File: js-packages/@prestojs/ui-antd/src/getWidgetForField.ts [hunk 1]
```diff
@@ -171,6 +169,7 @@ export default function getWidgetForField<
             const finalProps = {
                 ...props,
                 ...extraProps,
+                ..._field.getWidgetProps(),
```
Comment (fanglong-alliance) [inline js-packages/@prestojs/ui-antd/src/getWidgetForField.ts]: should it be the last here or first (or at least before extraProps)?

File: js-packages/@prestojs/ui/src/getFormatterForField.ts [hunk 2]
```diff
@@ -99,11 +97,14 @@ export default function getFormatterForField<
...
         if (f.choices) {
             if (Array.isArray(w)) {
-                return [w[0], { ...w[1], choices: f.choices }];
+                return [w[0], { ...w[1], choices: f.choices, ...f.getFormatterProps() }];
             } else {
-                return [w, { choices: f.choices }];
+                return [w, { choices: f.choices, ...f.getFormatterProps() }];
             }
         } else {
+            if (w) {
```
Comment (fanglong-alliance) [inline js-packages/@prestojs/ui/src/getFormatterForField.ts]: hmm, when would w be falsy? it also looks like its assumed if f is a choices

Comment (davecoates) [reply js-packages/@prestojs/ui/src/getFormatterForField.ts]: hmm yeah looks like you are right - i'll add missing types to make it more obvious

File: js-packages/@prestojs/ui-antd/src/getWidgetForField.ts (hunk 1 above)
Comment (davecoates) [reply js-packages/@prestojs/ui-antd/src/getWidgetForField.ts]: ah yeah good one - should come first so can be overriden

**Pull request merged**